util/src/test/java/tech/pegasys/teku/util/config/KeyStoreFilesLocatorTest.java

```diff
@@ -0,0 +1,168 @@
+    assertThat(locator.getFilePairs()).containsExactly(tuple(tempStr, "key/a.json", "pass/a.txt"));
+  }
+
+  @Test
+  public void shouldHandleFilesAndFoldersInOneArgument(@TempDir final Path tempDir)
+      throws IOException {
+    createFolders(tempDir, "key", "pass");
+    createFiles(tempDir, "key/a", "pass/a.txt", "keyStore", "password");
+    final String tempStr = tempDir.toString();
+    KeyStoreFilesLocator locator =
+        new KeyStoreFilesLocator(
+            Optional.of(
+                List.of(
+                    String.format("%s/key/a:%s/pass/a.txt", tempStr, tempStr),
+                    String.format("%s/keyStore:%s/password", tempStr, tempStr))),
+            ":");
+    locator.parse();
+    assertThat(locator.getFilePairs().size()).isEqualTo(2);
+  }
+
+  @Test
+  public void shouldDetectMissingPasswordFileWhenDirectoryIsPresent(@TempDir final Path tempDir)
+      throws IOException {
+    createFolders(tempDir, "key", "pass/a.txt");
+    createFiles(tempDir, "key/a");
+    final String tempStr = tempDir.toString();
+    KeyStoreFilesLocator locator =
+        new KeyStoreFilesLocator(
+            Optional.of(List.of(String.format("%s/key/a;%s/pass/a.txt", tempStr, tempStr))), ";");
+    assertThatThrownBy(locator::parse).isInstanceOf(InvalidConfigurationException.class);
+  }
+
+  @Test
+  public void shouldDetectMissingPasswordFile(@TempDir final Path tempDir) throws IOException {
+    createFolders(tempDir, "key", "pass");
+    createFiles(tempDir, "pass/a.txt");
+    final String tempStr = tempDir.toString();
+    KeyStoreFilesLocator locator =
+        new KeyStoreFilesLocator(
+            Optional.of(List.of(String.format("%s/key/a:%s/pass/a.txt", tempStr, tempStr))), ":");
+    assertThatThrownBy(locator::parse).isInstanceOf(InvalidConfigurationException.class);
+  }
+
+  @Test
+  public void shouldDetectMissingKeyFile(@TempDir final Path tempDir) throws IOException {
+    createFolders(tempDir, "key", "pass");
+    createFiles(tempDir, "key/a");
+    final String tempStr = tempDir.toString();
+    KeyStoreFilesLocator locator =
+        new KeyStoreFilesLocator(
+            Optional.of(List.of(String.format("%s/key/a:%s/pass/a.txt", tempStr, tempStr))), ":");
+    assertThatThrownBy(locator::parse).isInstanceOf(InvalidConfigurationException.class);
+  }
+
+  @Test
+  public void shouldSucceedCallingParseOnEmptyList() {
+    KeyStoreFilesLocator locator = new KeyStoreFilesLocator(Optional.empty(), ":");
+    locator.parse();
+    assertThat(locator.getFilePairs()).isEmpty();
+  }
+
+  @Test
+  public void shouldHandleOldArgs(@TempDir final Path tempDir) throws IOException {
+    createFolders(tempDir, "key", "pass");
+    createFiles(tempDir, "key/a", "pass/a.txt");
+    final String tempStr = tempDir.toString();
+    KeyStoreFilesLocator locator = new KeyStoreFilesLocator(Optional.empty(), ":");
+    locator.parseKeyAndPasswordList(List.of(tempStr + "/key/a"), List.of(tempStr + "/pass/a.txt"));
```

Comment:
Need to avoid hard coding `/` as the path separator and use `Path.resolve` or `Path.of` to join the pieces.